### ISSUE TITLE
Elden Ring DLC update broke .dcx decompilation

### ISSUE THREAD

**Issue by FabulaUltimaSheet** [thefifthmatt/ESDLang#5]
The 21/06/24 update for Elden Ring changed the regulation.bin file, and the tool doesn't decompile properly the .dcx files for dialogues

**Comment by gunnm730** [comment]
i m having the same issue.and i cant find another tool can help

**Comment by wgodkill** [comment]
Yep, I guess we might need a new tool or pray for the dear author to update

**Comment by bossman90** [comment]
Same!

**Comment by Triple-S** [comment]
OK, so apparently I'm not doing something wrong like I initially assumed, its rather the tool that is currently not working. Would be really great if someone could fix this.

**Comment by gracenotes** [comment]
A temporary workaround for this is to temporarily back up regulation.bin in your game directory, replace it with a pre-DLC version like 1.10 (can be download from https://www.nexusmods.com/eldenring/mods/4262), and proceed with decompilation. You can restore the backup to play the game normally.

An ESDLang update is in the works, but it needs more than just dcx-reading to be compatible with DLC. There is also additional support needed for extracting DLC-specific metadata and incorporating ESDStudio's backwards-incompatible changes to the language itself. Also, some options to make this issue not happen again in the future, such as a mod directories and making metadata optional.

**Comment by gracenotes** [comment]
As of v0.4.1 I've updated to SoulsFormatsNEXT which has support for all dcx formats (although its PARAM.cs is broken for AC6, which may require a pull request from me). I've also added a bunch of command names and enums and reading DLC names correctly, basically clearing out nearly the entire feature backlog. Apologies it took this long, as DLC randomizer took an unexpectedly long time, and then burnout from DLC randomizer.
